--- libvirt-6.6.0/docs/Makefile.am~	2020-07-21 15:34:44.000000000 +0200
+++ libvirt-6.6.0/docs/Makefile.am	2022-06-12 10:36:19.495506442 +0200
@@ -411,17 +411,17 @@
 	 grep -v '^:Manual ' < $< | \
 	  sed -e 's|SYSCONFDIR|$(sysconfdir)|g' \
 	     -e 's|RUNSTATEDIR|$(runstatedir)|g' | \
-	  $(RST2HTML) --strict > $@ || { rm $@ && exit 1; }
+	  $(RST2HTML) --stylesheet-path=minimal.css,responsive.css --strict > $@ || { rm $@ && exit 1; }
 
 news.html.in: $(top_srcdir)/NEWS.rst
 	$(AM_V_GEN)$(MKDIR_P) `dirname $@` && \
-	  $(RST2HTML) --strict $< > $@ || { rm $@ && exit 1; }
+	  $(RST2HTML) --stylesheet-path=minimal.css,responsive.css --strict $< > $@ || { rm $@ && exit 1; }
 
 CLEANFILES += news.html.in
 
 %.html.in: %.rst
 	$(AM_V_GEN)$(MKDIR_P) `dirname $@` && \
-	  $(RST2HTML) --strict $< > $@ || { rm $@ && exit 1; }
+	  $(RST2HTML) --stylesheet-path=minimal.css,responsive.css --strict $< > $@ || { rm $@ && exit 1; }
 
 %.html.tmp: %.html.in site.xsl subsite.xsl page.xsl \
 		$(acl_generated)
