--- libvirt-6.6.0/src/qemu/qemu_command.c~	2020-07-21 13:39:30.000000000 +0000
+++ libvirt-6.6.0/src/qemu/qemu_command.c	2021-07-08 18:33:22.029690230 +0000
@@ -7644,6 +7644,9 @@
         /* TODO: Support ACLs later */
     }
 
+    if (cfg->vncACL)
+	    virBufferAddLit(&opt, ",acl");
+
     virCommandAddArg(cmd, "-vnc");
     virCommandAddArgBuffer(cmd, &opt);
     if (graphics->data.vnc.keymap)

--- libvirt-6.6.0/src/qemu/qemu_conf.c~	2020-07-21 13:34:45.000000000 +0000
+++ libvirt-6.6.0/src/qemu/qemu_conf.c	2021-07-08 18:37:21.789487295 +0000
@@ -439,6 +439,8 @@
         return -1;
     if (virConfGetValueBool(conf, "vnc_allow_host_audio", &cfg->vncAllowHostAudio) < 0)
         return -1;
+    if (virConfGetValueBool(conf, "vnc_acl", &cfg->vncACL) < 0)
+	return -1;
 
     return 0;
 }

--- libvirt-6.6.0/src/qemu/qemu_conf.h~	2020-07-21 13:34:45.000000000 +0000
+++ libvirt-6.6.0/src/qemu/qemu_conf.h	2021-07-08 18:38:13.950893081 +0000
@@ -119,6 +119,7 @@
     bool vncTLSx509verify;
     bool vncTLSx509verifyPresent;
     bool vncSASL;
+    bool vncACL;
     char *vncTLSx509certdir;
     char *vncTLSx509secretUUID;
     char *vncListen;

--- libvirt-6.6.0/src/qemu/qemu.conf~	2021-07-08 18:39:03.982241321 +0000
+++ libvirt-6.6.0/src/qemu/qemu.conf	2021-07-08 18:42:35.567942073 +0000
@@ -141,6 +141,15 @@
 #vnc_sasl = 1
 
 
+# Enable the VNC access control lists. When switched on this will
+# initially block all vnc users from accessing the vnc server. To
+# add and remove ids from the ACLs you will need to send the appropriate
+# commands to the qemu monitor as required by your particular version of
+# QEMU. See the QEMU documentation for more details.
+#
+# vnc_acl = 1
+
+
 # The default SASL configuration file is located in /etc/sasl/
 # When running libvirtd unprivileged, it may be desirable to
 # override the configs in this location. Set this parameter to
