--- libvirt-6.10.0/src/meson.build.orig	2020-12-01 09:51:29.000000000 +0100
+++ libvirt-6.10.0/src/meson.build	2022-06-14 22:25:40.023912258 +0200
@@ -781,7 +781,7 @@ endforeach
 if conf.has('WITH_LIBVIRTD')
   # Generate systemd service and socket unit files
   if init_script == 'systemd'
-    systemd_unit_dir = prefix / 'lib' / 'systemd' / 'system'
+    systemd_unit_dir = '/lib' / 'systemd' / 'system'
 
     install_data(
       guest_unit_files,
--- libvirt-8.4.0/tools/meson.build~	2022-06-01 09:28:24.000000000 +0200
+++ libvirt-8.4.0/tools/meson.build	2022-06-21 08:43:36.653980962 +0200
@@ -315,7 +315,7 @@
       output: '@BASENAME@',
       configuration: tools_conf,
       install: true,
-      install_dir: prefix / 'lib' / 'systemd' / 'system',
+      install_dir: '/lib' / 'systemd' / 'system',
     )
   endif
 endif
