--- libvirt-1.2.4/src/xen/xen_driver.c.orig	2014-05-25 21:03:34.237326121 +0200
+++ libvirt-1.2.4/src/xen/xen_driver.c	2014-05-25 21:25:26.260632109 +0200
@@ -320,7 +320,7 @@
     if (virFileExists("/usr/sbin/xend")) {
         virCommandPtr cmd;
 
-        cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+        cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
         if (virCommandRun(cmd, NULL) == 0)
             ret = true;
         virCommandFree(cmd);
--- libvirt-1.2.4/src/libxl/libxl_driver.c.orig	2014-05-25 21:03:34.447326117 +0200
+++ libvirt-1.2.4/src/libxl/libxl_driver.c	2014-05-25 21:28:23.317295096 +0200
@@ -258,7 +258,7 @@
     if (virFileExists("/usr/sbin/xend")) {
         virCommandPtr cmd;
 
-        cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+        cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
         if (virCommandRun(cmd, NULL) == 0) {
             VIR_INFO("Legacy xen tool stack seems to be in use, disabling "
                      "libxenlight driver.");
--- libvirt-1.2.3/tests/statstest.c.orig	2014-04-04 15:57:05.109436604 +0200
+++ libvirt-1.2.3/tests/statstest.c	2014-04-04 16:00:09.836099420 +0200
@@ -49,7 +49,7 @@
     uname(&ut);
     if (strstr(ut.release, "xen") == NULL)
         return EXIT_AM_SKIP;
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, NULL) < 0) {
         virCommandFree(cmd);
         return EXIT_AM_SKIP;
--- libvirt-1.2.3/tests/reconnect.c.orig	2014-04-04 15:57:05.109436604 +0200
+++ libvirt-1.2.3/tests/reconnect.c	2014-04-04 16:03:28.892761936 +0200
@@ -24,7 +24,7 @@
     uname(&ut);
     if (strstr(ut.release, "xen") == NULL)
         return EXIT_AM_SKIP;
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, NULL) < 0) {
         virCommandFree(cmd);
         return EXIT_AM_SKIP;
