diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/daemon/libvirtd.conf libvirt-2.5.0/daemon/libvirtd.conf
--- libvirt-2.5.0.orig/daemon/libvirtd.conf	2016-06-27 16:12:20.000000000 +0200
+++ libvirt-2.5.0/daemon/libvirtd.conf	2016-12-20 12:56:04.000000000 +0100
@@ -123,7 +123,7 @@
 #          the network providing auth (eg, TLS/x509 certificates)
 #
 #  - sasl: use SASL infrastructure. The actual auth scheme is then
-#          controlled from /etc/sasl2/libvirt.conf. For the TCP
+#          controlled from /etc/sasl/libvirt.conf. For the TCP
 #          socket only GSSAPI & DIGEST-MD5 mechanisms will be used.
 #          For non-TCP or TLS sockets, any scheme is allowed.
 #
@@ -154,7 +154,7 @@
 # If you don't enable SASL, then all TCP traffic is cleartext.
 # Don't do this outside of a dev/test scenario. For real world
 # use, always enable SASL and use the GSSAPI or DIGEST-MD5
-# mechanism in /etc/sasl2/libvirt.conf
+# mechanism in /etc/sasl/libvirt.conf
 #auth_tcp = "sasl"
 
 # Change the authentication scheme for TLS sockets.
diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/daemon/Makefile.am libvirt-2.5.0/daemon/Makefile.am
--- libvirt-2.5.0.orig/daemon/Makefile.am	2016-11-30 13:57:30.000000000 +0100
+++ libvirt-2.5.0/daemon/Makefile.am	2016-12-20 12:56:04.000000000 +0100
@@ -532,13 +532,13 @@
 # the WITH_LIBVIRTD conditional
 if WITH_SASL
 install-data-sasl:
-	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/sasl2/
+	$(MKDIR_P) $(DESTDIR)$(sysconfdir)/sasl/
 	$(INSTALL_DATA) $(srcdir)/libvirtd.sasl \
-		$(DESTDIR)$(sysconfdir)/sasl2/libvirt.conf
+		$(DESTDIR)$(sysconfdir)/sasl/libvirt.conf
 
 uninstall-data-sasl:
-	rm -f $(DESTDIR)$(sysconfdir)/sasl2/libvirt.conf
-	rmdir $(DESTDIR)$(sysconfdir)/sasl2/ || :
+	rm -f $(DESTDIR)$(sysconfdir)/sasl/libvirt.conf
+	rmdir $(DESTDIR)$(sysconfdir)/sasl/ || :
 else ! WITH_SASL
 install-data-sasl:
 uninstall-data-sasl:
diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/docs/auth.html libvirt-2.5.0/docs/auth.html
--- libvirt-2.5.0.orig/docs/auth.html	2016-11-29 10:50:11.000000000 +0100
+++ libvirt-2.5.0/docs/auth.html	2016-12-20 12:56:04.000000000 +0100
@@ -257,7 +257,7 @@
 The plain TCP socket of the libvirt daemon defaults to using SASL for authentication.
 The SASL mechanism configured by default is DIGEST-MD5, which provides a basic
 username+password style authentication. To enable Kerberos single-sign-on instead,
-the libvirt SASL configuration file must be changed. This is <code>/etc/sasl2/libvirt.conf</code>.
+the libvirt SASL configuration file must be changed. This is <code>/etc/sasl/libvirt.conf</code>.
 The <code>mech_list</code> parameter must first be changed to <code>gssapi</code>
 instead of the default <code>digest-md5</code>, and keytab should be set to
 <code>/etc/libvirt/krb5.tab</code> . If SASL is enabled on the UNIX
diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/docs/auth.html.in libvirt-2.5.0/docs/auth.html.in
--- libvirt-2.5.0.orig/docs/auth.html.in	2016-06-26 10:01:24.000000000 +0200
+++ libvirt-2.5.0/docs/auth.html.in	2016-12-20 12:56:04.000000000 +0100
@@ -246,7 +246,7 @@
 The plain TCP socket of the libvirt daemon defaults to using SASL for authentication.
 The SASL mechanism configured by default is DIGEST-MD5, which provides a basic
 username+password style authentication. To enable Kerberos single-sign-on instead,
-the libvirt SASL configuration file must be changed. This is <code>/etc/sasl2/libvirt.conf</code>.
+the libvirt SASL configuration file must be changed. This is <code>/etc/sasl/libvirt.conf</code>.
 The <code>mech_list</code> parameter must first be changed to <code>gssapi</code>
 instead of the default <code>digest-md5</code>, and keytab should be set to
 <code>/etc/libvirt/krb5.tab</code> . If SASL is enabled on the UNIX
diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/src/qemu/qemu.conf libvirt-2.5.0/src/qemu/qemu.conf
--- libvirt-2.5.0.orig/src/qemu/qemu.conf	2016-11-29 10:47:58.000000000 +0100
+++ libvirt-2.5.0/src/qemu/qemu.conf	2016-12-20 12:56:04.000000000 +0100
@@ -112,18 +112,18 @@
 # Examples include vinagre, virt-viewer and virt-manager
 # itself. UltraVNC, RealVNC, TightVNC do not support this
 #
-# It is necessary to configure /etc/sasl2/qemu.conf to choose
+# It is necessary to configure /etc/sasl/qemu.conf to choose
 # the desired SASL plugin (eg, GSSPI for Kerberos)
 #
 #vnc_sasl = 1
 
 
-# The default SASL configuration file is located in /etc/sasl2/
+# The default SASL configuration file is located in /etc/sasl/
 # When running libvirtd unprivileged, it may be desirable to
 # override the configs in this location. Set this parameter to
 # point to the directory, and create a qemu.conf in that location
 #
-#vnc_sasl_dir = "/some/directory/sasl2"
+#vnc_sasl_dir = "/some/directory/sasl"
 
 
 # QEMU implements an extension for providing audio over a VNC connection,
diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/tests/qemuxml2argvdata/qemuxml2argv-graphics-vnc-sasl.args libvirt-2.5.0/tests/qemuxml2argvdata/qemuxml2argv-graphics-vnc-sasl.args
--- libvirt-2.5.0.orig/tests/qemuxml2argvdata/qemuxml2argv-graphics-vnc-sasl.args	2016-07-27 14:39:35.000000000 +0200
+++ libvirt-2.5.0/tests/qemuxml2argvdata/qemuxml2argv-graphics-vnc-sasl.args	2016-12-20 12:56:04.000000000 +0100
@@ -3,7 +3,7 @@
 HOME=/home/test \
 USER=test \
 LOGNAME=test \
-SASL_CONF_PATH=/root/.sasl2 \
+SASL_CONF_PATH=/root/.sasl \
 QEMU_AUDIO_DRV=none \
 /usr/bin/qemu \
 -name QEMUGuest1 \
diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/tests/qemuxml2argvdata/qemuxml2argv-graphics-vnc-tls.args libvirt-2.5.0/tests/qemuxml2argvdata/qemuxml2argv-graphics-vnc-tls.args
--- libvirt-2.5.0.orig/tests/qemuxml2argvdata/qemuxml2argv-graphics-vnc-tls.args	2016-07-27 14:39:35.000000000 +0200
+++ libvirt-2.5.0/tests/qemuxml2argvdata/qemuxml2argv-graphics-vnc-tls.args	2016-12-20 12:56:04.000000000 +0100
@@ -3,7 +3,7 @@
 HOME=/home/test \
 USER=test \
 LOGNAME=test \
-SASL_CONF_PATH=/root/.sasl2 \
+SASL_CONF_PATH=/root/.sasl \
 QEMU_AUDIO_DRV=none \
 /usr/bin/qemu \
 -name QEMUGuest1 \
diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/tests/virconfdata/libvirtd.conf libvirt-2.5.0/tests/virconfdata/libvirtd.conf
--- libvirt-2.5.0.orig/tests/virconfdata/libvirtd.conf	2016-04-26 04:20:09.000000000 +0200
+++ libvirt-2.5.0/tests/virconfdata/libvirtd.conf	2016-12-20 12:56:04.000000000 +0100
@@ -108,7 +108,7 @@
 #          the network providing auth (eg, TLS/x509 certificates)
 #
 #  - sasl: use SASL infrastructure. The actual auth scheme is then
-#          controlled from /etc/sasl2/libvirt.conf. For the TCP
+#          controlled from /etc/sasl/libvirt.conf. For the TCP
 #          socket only GSSAPI & DIGEST-MD5 mechanisms will be used.
 #          For non-TCP or TLS sockets,  any scheme is allowed.
 #
@@ -139,7 +139,7 @@
 # If you don't enable SASL, then all TCP traffic is cleartext.
 # Don't do this outside of a dev/test scenario. For real world
 # use, always enable SASL and use the GSSAPI or DIGEST-MD5
-# mechanism in /etc/sasl2/libvirt.conf
+# mechanism in /etc/sasl/libvirt.conf
 auth_tcp = "sasl"
 
 # Change the authentication scheme for TLS sockets.
diff -dur -x '*~' -x '*.orig' -x '*.rej' libvirt-2.5.0.orig/tests/virconfdata/libvirtd.out libvirt-2.5.0/tests/virconfdata/libvirtd.out
--- libvirt-2.5.0.orig/tests/virconfdata/libvirtd.out	2016-04-26 04:20:09.000000000 +0200
+++ libvirt-2.5.0/tests/virconfdata/libvirtd.out	2016-12-20 12:56:04.000000000 +0100
@@ -87,7 +87,7 @@
 #          the network providing auth (eg, TLS/x509 certificates)
 #
 #  - sasl: use SASL infrastructure. The actual auth scheme is then
-#          controlled from /etc/sasl2/libvirt.conf. For the TCP
+#          controlled from /etc/sasl/libvirt.conf. For the TCP
 #          socket only GSSAPI & DIGEST-MD5 mechanisms will be used.
 #          For non-TCP or TLS sockets,  any scheme is allowed.
 #
@@ -116,7 +116,7 @@
 # If you don't enable SASL, then all TCP traffic is cleartext.
 # Don't do this outside of a dev/test scenario. For real world
 # use, always enable SASL and use the GSSAPI or DIGEST-MD5
-# mechanism in /etc/sasl2/libvirt.conf
+# mechanism in /etc/sasl/libvirt.conf
 auth_tcp = "sasl"
 # Change the authentication scheme for TLS sockets.
 #
