--- libvirt-0.9.10/src/xen/xen_driver.c~	2012-02-13 08:06:23.000000000 +0100
+++ libvirt-0.9.10/src/xen/xen_driver.c	2012-03-06 14:03:45.273637374 +0100
@@ -240,7 +240,7 @@
     int status;
     int ret = 0;
 
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, &status) == 0 && status == 0)
         ret = 1;
     virCommandFree(cmd);
--- libvirt-1.1.3/src/libxl/libxl_driver.c.orig	2013-10-22 20:46:08.195357964 +0200
+++ libvirt-1.1.3/src/libxl/libxl_driver.c	2013-10-22 21:17:55.935277904 +0200
@@ -797,7 +797,7 @@
     }
 
     /* Don't load if legacy xen toolstack (xend) is in use */
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, &status) == 0 && status == 0) {
         VIR_INFO("Legacy xen tool stack seems to be in use, disabling "
                   "libxenlight driver.");
--- libvirt-0.9.10/tests/statstest.c~	2011-11-22 02:52:24.000000000 +0100
+++ libvirt-0.9.10/tests/statstest.c	2012-03-06 14:36:59.970394039 +0100
@@ -56,7 +56,7 @@
     uname(&ut);
     if (strstr(ut.release, "xen") == NULL)
         return EXIT_AM_SKIP;
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, &status) != 0 || status != 0) {
         virCommandFree(cmd);
         return EXIT_AM_SKIP;
--- libvirt-0.9.10/tests/reconnect.c~	2011-11-22 02:52:24.000000000 +0100
+++ libvirt-0.9.10/tests/reconnect.c	2012-03-06 14:37:27.777061955 +0100
@@ -29,7 +29,7 @@
     uname(&ut);
     if (strstr(ut.release, "xen") == NULL)
         return EXIT_AM_SKIP;
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, &status) != 0 || status != 0) {
         virCommandFree(cmd);
         return EXIT_AM_SKIP;
