--- libvirt-6.4.0/m4/virt-bash-completion.m4.orig	2019-11-26 22:17:34.576588169 +0100
+++ libvirt-6.4.0/m4/virt-bash-completion.m4	2020-06-21 11:00:05.173640478 +0200
@@ -37,11 +37,12 @@
     fi
   fi
 
-  LIBVIRT_CHECK_PKG([BASH_COMPLETION], [bash-completion], [2.0])
+  AM_CONDITIONAL([WITH_BASH_COMPLETION], [false])
 
   if test "x$with_bash_completion" = "xyes" ; then
     if test "x$with_bash_completions_dir" = "xcheck"; then
       AC_MSG_CHECKING([for bash-completions directory])
+      LIBVIRT_CHECK_PKG([BASH_COMPLETION], [bash-completion], [2.0])
       BASH_COMPLETIONS_DIR="$($PKG_CONFIG --variable=completionsdir bash-completion)"
       AC_MSG_RESULT([$BASH_COMPLETIONS_DIR])
 
@@ -59,6 +60,7 @@
       AC_MSG_ERROR([bash-completions-dir must be used only with valid path])
     else
       BASH_COMPLETIONS_DIR=$with_bash_completions_dir
+      AM_CONDITIONAL([WITH_BASH_COMPLETION], [true])
     fi
     AC_SUBST([BASH_COMPLETIONS_DIR])
   fi
