diff -urNp -x '*.orig' libvirt-8.4.0.org/src/qemu/qemu.conf.in libvirt-8.4.0/src/qemu/qemu.conf.in
--- libvirt-8.4.0.org/src/qemu/qemu.conf.in	2022-06-19 22:15:27.653173791 +0200
+++ libvirt-8.4.0/src/qemu/qemu.conf.in	2022-06-19 22:15:28.366510311 +0200
@@ -147,6 +147,15 @@
 #vnc_sasl = 1
 
 
+# Enable the VNC access control lists. When switched on this will
+# initially block all vnc users from accessing the vnc server. To
+# add and remove ids from the ACLs you will need to send the appropriate
+# commands to the qemu monitor as required by your particular version of
+# QEMU. See the QEMU documentation for more details.
+#
+# vnc_acl = 1
+
+
 # The default SASL configuration file is located in /etc/sasl/
 # When running libvirtd unprivileged, it may be desirable to
 # override the configs in this location. Set this parameter to
--- libvirt-8.8.0/src/qemu/qemu_command.c.orig	2022-10-09 21:14:42.031046895 +0200
+++ libvirt-8.8.0/src/qemu/qemu_command.c	2022-10-09 21:16:56.026987643 +0200
@@ -7737,6 +7737,9 @@ qemuBuildGraphicsVNCCommandLine(virQEMUD
 
     virBufferAsprintf(&opt, ",audiodev=%s", audioid);
 
+    if (cfg->vncACL)
+	    virBufferAddLit(&opt, ",acl");
+
     virCommandAddArg(cmd, "-vnc");
     virCommandAddArgBuffer(cmd, &opt);
     if (graphics->data.vnc.keymap)
diff -urNp -x '*.orig' libvirt-8.4.0.org/src/qemu/qemu_conf.c libvirt-8.4.0/src/qemu/qemu_conf.c
--- libvirt-8.4.0.org/src/qemu/qemu_conf.c	2022-06-01 09:28:24.000000000 +0200
+++ libvirt-8.4.0/src/qemu/qemu_conf.c	2022-06-19 22:15:28.366510311 +0200
@@ -439,6 +439,8 @@ virQEMUDriverConfigLoadVNCEntry(virQEMUD
         return -1;
     if (virConfGetValueBool(conf, "vnc_allow_host_audio", &cfg->vncAllowHostAudio) < 0)
         return -1;
+    if (virConfGetValueBool(conf, "vnc_acl", &cfg->vncACL) < 0)
+	return -1;
 
     if (cfg->vncPassword &&
         strlen(cfg->vncPassword) > 8) {
diff -urNp -x '*.orig' libvirt-8.4.0.org/src/qemu/qemu_conf.h libvirt-8.4.0/src/qemu/qemu_conf.h
--- libvirt-8.4.0.org/src/qemu/qemu_conf.h	2022-06-01 09:28:24.000000000 +0200
+++ libvirt-8.4.0/src/qemu/qemu_conf.h	2022-06-19 22:15:28.366510311 +0200
@@ -114,6 +114,7 @@ struct _virQEMUDriverConfig {
     bool vncTLSx509verify;
     bool vncTLSx509verifyPresent;
     bool vncSASL;
+    bool vncACL;
     char *vncTLSx509certdir;
     char *vncTLSx509secretUUID;
     char *vncListen;
