--- libvirt-5.4.0/src/util/virutil.c.orig	2019-06-03 18:28:22.409062013 +0200
+++ libvirt-5.4.0/src/util/virutil.c	2019-06-03 18:29:23.278797431 +0200
@@ -1488,7 +1488,7 @@
     if (!(udev = virFindFileInPath(UDEVADM)))
         return;
 
-    if (!(cmd = virCommandNewArgList(udev, "settle", NULL)))
+    if (!(cmd = virCommandNewArgList(udev, "settle", "--timeout=10", NULL)))
         return;
 
     /*
