--- libvirt-1.2.3/src/xen/xen_driver.c.orig	2014-04-04 15:57:05.069436605 +0200
+++ libvirt-1.2.3/src/xen/xen_driver.c	2014-04-04 15:58:59.119434226 +0200
@@ -318,7 +318,7 @@
     virCommandPtr cmd;
     bool ret = false;
 
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, NULL) == 0)
         ret = true;
     virCommandFree(cmd);
--- libvirt-1.1.3/src/libxl/libxl_driver.c.orig	2013-10-22 20:46:08.195357964 +0200
+++ libvirt-1.1.3/src/libxl/libxl_driver.c	2013-10-22 21:17:55.935277904 +0200
@@ -797,7 +797,7 @@
     }
 
     /* Don't load if legacy xen toolstack (xend) is in use */
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, &status) == 0 && status == 0) {
         VIR_INFO("Legacy xen tool stack seems to be in use, disabling "
                   "libxenlight driver.");
--- libvirt-1.2.3/tests/statstest.c.orig	2014-04-04 15:57:05.109436604 +0200
+++ libvirt-1.2.3/tests/statstest.c	2014-04-04 16:00:09.836099420 +0200
@@ -49,7 +49,7 @@
     uname(&ut);
     if (strstr(ut.release, "xen") == NULL)
         return EXIT_AM_SKIP;
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, NULL) < 0) {
         virCommandFree(cmd);
         return EXIT_AM_SKIP;
--- libvirt-1.2.3/tests/reconnect.c.orig	2014-04-04 15:57:05.109436604 +0200
+++ libvirt-1.2.3/tests/reconnect.c	2014-04-04 16:03:28.892761936 +0200
@@ -24,7 +24,7 @@
     uname(&ut);
     if (strstr(ut.release, "xen") == NULL)
         return EXIT_AM_SKIP;
-    cmd = virCommandNewArgList("/usr/sbin/xend", "status", NULL);
+    cmd = virCommandNewArgList("/sbin/service", "xend", "status", NULL);
     if (virCommandRun(cmd, NULL) < 0) {
         virCommandFree(cmd);
         return EXIT_AM_SKIP;
