--- libvirt-6.10.0/src/meson.build.orig	2020-12-01 09:51:29.000000000 +0100
+++ libvirt-6.10.0/src/meson.build	2022-06-14 22:25:40.023912258 +0200
@@ -781,7 +781,7 @@ endforeach
 if conf.has('WITH_LIBVIRTD')
   # Generate systemd service and socket unit files
   if init_script == 'systemd'
-    systemd_unit_dir = prefix / 'lib' / 'systemd' / 'system'
+    systemd_unit_dir = '/lib' / 'systemd' / 'system'
 
     install_data(
       guest_unit_files,
