--- libvirt-0.9.10/src/lxc/lxc_container.c~	2012-02-13 08:06:23.000000000 +0100
+++ libvirt-0.9.10/src/lxc/lxc_container.c	2012-03-17 20:40:10.960883236 +0100
@@ -501,7 +501,7 @@
     if (pivotRoot) {
 #if HAVE_SELINUX
         if (getfilecon("/", &con) < 0 &&
-            errno != ENOTSUP) {
+            errno != ENOTSUP && errno != ENODATA) {
             virReportSystemError(errno, "%s",
                                  _("Failed to query file context on /"));
             goto cleanup;
--- libvirt-0.9.10/src/lxc/lxc_controller.c~	2012-03-17 20:14:16.117479754 +0100
+++ libvirt-0.9.10/src/lxc/lxc_controller.c	2012-03-17 20:41:17.730886249 +0100
@@ -1492,7 +1492,7 @@
 
 #if HAVE_SELINUX
         if (getfilecon(root->src, &con) < 0 &&
-            errno != ENOTSUP) {
+            errno != ENOTSUP && errno != ENODATA) {
             virReportSystemError(errno,
                                  _("Failed to query file context on %s"),
                                  root->src);
